use equivalentClass instead of AbstractClass

ilans opened the pull request:
An alternative approach for flagging a class as abstract, instead of subclassing it with "http://spdx.invalid./AbstractClass".
This approach adds SHACL validation that prevents abstract classes from being directly instantiated.
It looks something like:
```
:ExtendableLicense a owl:Class ;
    sh:not [ 
        sh:class :ExtendableLicense ;
        sh:message "ExtendableLicense is an abstract class and should not be instantiated directly. Instantiate a subclass instead."@en 
    ] .
```

goneall commented:
@zvr @JPEWdev - Pls review - it looks like this should work, but it requires changes to the code generators as well as the spec parser.  I have no problem updating the Java code generator if this approach works for both of you.

ilans commented:
I've updated the PR with a diffetent method that doesn't require listing subclasses in the constraints.
Pleasse read the updated description.

ilans commented:
Changed the PR to use a simpler method:
- Removed OWL.oneOf because it's not needed for validation.
- Instead of SH.disjoint I'm using a more straightforward approach with SH.not on the SH.class.
- Added a helpful SH.message.
- Updated the PR description with an example.

For validation, I run this script:
```python
import rdflib
from rdflib import SH
from pyshacl import validate

s = """
@prefix ex: <http://example.org/ns#> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix spdxcore: <https://spdx.org/rdf/3.0.1/terms/Core/> .

ex:MyAgent
    a spdxcore:Agent ;
    spdxcore:creationInfo _:MyCreationInfo .

_:MyCreationInfo
    a spdxcore:CreationInfo ;
    spdxcore:createdBy ex:MyAgent ;
    spdxcore:created "2024-09-04T20:25:34Z"^^xsd:dateTimeStamp ;
    spdxcore:specVersion "3.0.1" .

ex:MyElement
    a spdxcore:Element ;
    spdxcore:creationInfo _:MyCreationInfo .

ex:MyTool
    a spdxcore:Tool ;
    spdxcore:creationInfo _:MyCreationInfo .

"""
data_graph = rdflib.Graph()
data_graph.parse(data=s, format="turtle")

shapes_graph = rdflib.Graph()
shapes_graph.parse("output/rdf/spdx-model.ttl", format="turtle") 

_, report_graph, _ = validate(data_graph, shacl_graph=shapes_graph, allow_warnings=True)
for violation in report_graph.subjects(rdflib.RDF.type, SH["ValidationResult"]):
    message = report_graph.value(subject=violation, predicate=SH["resultMessage"], )
    print(message)
```

Which produces:
```
Node ex:MyElement must not conform to shape [ 
  sh:class ns1:Element ; 
  sh:message Literal(
    "https://spdx.org/rdf/3.0.1/terms/Core/Element is an abstract class and should not be instantiated directly. Instantiate a subclass instead.", 
    lang=en
  ) 
]
```

JPEWdev commented:
Excellent. I like this solution. I'll work on adding support for this idiom to shacl2code. My only request here is that we squash the commits when they are merged (or squash them before merging in the PR)

zvr commented:
There are two changes here:

a. remove the Abstract class
b. implement what it means to be an abstract class in SHACL

I'm fully supporting (b) above -- we needed this!

Regarding (a), for people who use the TTL file to automatically produce class generations in some language, is everyone OK with not having the simple `Abstract` class? The discovery that a class is abstract in our model should then be made by recognizing the generated SHACL entries.

goneall commented:
> There are two changes here:
> 
> a. remove the Abstract class b. implement what it means to be an abstract class in SHACL
> 
> I'm fully supporting (b) above -- we needed this!
> 
> Regarding (a), for people who use the TTL file to automatically produce class generations in some language, is everyone OK with not having the simple `Abstract` class? The discovery that a class is abstract in our model should then be made by recognizing the generated SHACL entries.

I can update the Java code generator to work with this new approach.  It sounds like @JPEWdev is also cool with this.  I don't know of any other tools that are using this approach, so I think we're OK removing it.

JPEWdev commented:
https://github.com/JPEWdev/shacl2code/pull/28

goneall commented:
@zvr - Can we merge this?  I'm using the feature branch for my tooling and it would be great to use an official merged version.

goneall commented:
I believe this PR caused issue #165
